src/polymer/pseudo-element-scanner.ts

```diff
@@ -18,40 +18,82 @@ import {ASTNode} from 'parse5';
 import {ScannedPolymerElement} from './polymer-element';
 
+function parseComment(comment: string): ScannedPolymerElement|undefined {
+  const parsedJsdoc = jsdoc.parseJsdoc(comment);
+  const pseudoTag = jsdoc.getTag(parsedJsdoc, 'pseudoElement', 'name');
+  if (pseudoTag) {
+    let element = new ScannedPolymerElement({
+      tagName: pseudoTag,
+      jsdoc: {description: parsedJsdoc.description, tags: parsedJsdoc.tags},
+      properties: [],
+      description: parsedJsdoc.description,
+      sourceRange: undefined
+    });
+    element.pseudo = true;
+    annotateElementHeader(element);
+    return element;
+  }
+}
 /**
  * A Polymer pseudo-element is an element that is declared in an unusual way, such
  * that the analyzer couldn't normally analyze it, so instead it is declared in
  * comments.
  */
-export class PseudoElementScanner implements HtmlScanner {
-  async scan(
+export class PseudoElementScanner implements HtmlScanner, JavaScriptScanner {
+  async scanHtml(
       document: ParsedHtmlDocument,
       visit: (visitor: HtmlVisitor) => Promise<void>):
       Promise<ScannedPolymerElement[]> {
     let elements: ScannedPolymerElement[] = [];
 
     await visit((node: ASTNode) => {
       if (dom5.isCommentNode(node) && node.data && node.data.includes('@pseudoElement')) {
-        const parsedJsdoc = jsdoc.parseJsdoc(node.data);
-        const pseudoTag = jsdoc.getTag(parsedJsdoc, 'pseudoElement', 'name');
-        if (pseudoTag) {
-          let element = new ScannedPolymerElement({
-            tagName: pseudoTag,
-            jsdoc: {description: parsedJsdoc.description, tags: parsedJsdoc.tags},
-            properties: [],
-            description: parsedJsdoc.description,
-            sourceRange: document.sourceRangeForNode(node)
-          });
-          element.pseudo = true;
-          annotateElementHeader(element);
+        let element = parseComment(node.data);
+        if (element) {
+          element.sourceRange = document.sourceRangeForNode(node);
           elements.push(element);
         }
       }
     });
     return elements;
   }
+
+  async scanJs(document: JavaScriptDocument): Promise<ScannedPolymerElement[]> {
+    let elements: ScannedPolymerElement[] = [];
+
+    for (let comment of document.ast.comments) {
+        let element = parseComment(comment.value);
+        if (element) {
+          element.sourceRange = document.sourceRangeForNode(<Node>comment);
+          elements.push(element);
+        }
+    }
+
+    return elements;
+  }
+
+  async scan(
+      document: JavaScriptDocument, visit: (visitor: Visitor) => Promise<void>):
+      Promise<ScannedPolymerElement[]>;
+  async scan(
+      document: ParsedHtmlDocument,
+      visit: (visitor: HtmlVisitor) => Promise<void>):
+      Promise<ScannedPolymerElement[]>;
+  async scan(document: any, visit: any): Promise<any> {
+    if (document instanceof JavaScriptDocument) {
+      return this.scanJs(document);
```

Comment:
Since this scanner essentially has two nearly independent implementations, and by default there are two instances - one for HTML and one for JS - I'd just make two separate scanners. It'll be less complicated. `parseComment` can go in a shared module.